### ISSUE TITLE
Is there a way to define a schema for a function that returns void?

### ISSUE THREAD

**Issue by neodon** [vriad/zod#77]
First of all, zod is awesome.  I would recommend adding it to the awesome-typescript list(s) as an alternative to io-ts.

I'm trying to define a schema for a function that does not return a value.  Example:

```typescript
function foo(): void {}
```

Currently I am using `z.unknown` for `returnType`:

```typescript
const FooSchema = z.function(z.tuple([]), z.unknown())
```

This works, but is it the correct way to do it?  Should there be a void type for function schemas?

**Comment by vriad** [comment]
I've actually already implemented this but it's deeply buried in my zod2 branch which isn't released/stable yet :/ 

For now I recommend using this snippet. It's pretty horrible so sorry about that, but it is technically correct and will work with type inference:

```
const zvoid: z.ZodType<void> = z.union([z.undefined(), z.null()]) as any;
```

**Comment by neodon** [comment]
Thank you! This helps and I appreciate the quick response.

**Comment by vriad** [comment]
@neodon Just published zod@1.8.0 which includes `z.void()` 👍
